src/components/WalletModal/index.tsx

```diff
@@ -198,8 +141,8 @@ export default function WalletModal({ openSettings }: { openSettings: () => void
                 <Option
                   key={connection.getName()}
                   connection={connection}
-                  activate={() => tryActivation(connection)}
-                  pendingConnectionType={pendingConnection?.type}
+                  activate={() => activate(connection)}
+                  pendingConnectionType={pending.connection?.type}
```

Comment:
Could we have a follow-up to refactor this component to get rid of `pendingConnectionType` in favor of two new props, `pending` and `disabled`? That would make it more explicit than passing this prop. Wasn't initially clear to me what's the purpose of it.

Reply:
Sounds good, agree this is confusing